Dual OTR MQ and Vanilla Support

leggettc18 opened the pull request:
The goal of this PR is to enable Loading from two OTRs, which will enable switching between Vanilla and MQ in ImGui and Randomizer setting random dungeons to be their Master Quest layouts.

Requirements:

- [x] OTR paths to scenes are different depending on whether or not the OTR was extracted from a Master Quest rom.
- [x] When only one OTR is available, SoH modifies which path it uses for loading scenes depending on whether or not the currently loaded OTR was extracted from a Master Quest rom.
- [x] When both a Vanilla and Master Quest OTR are available, SoH loads the Master Quest OTR as the base and the Vanilla OTR as a patch, so all the shared content comes from the Vanilla OTR and dungeons can optionally come from Master Quest OTR, depending on settings.
- [x] If both OTRs are loaded, Master Quest can be enabled or disabled from ImGui. But this setting can only be changed while on the File Select screen. Loading an existing save will enable or disable this setting according to whether it was created with Master Quest enabled or not (similar to rando).
- [x] If both OTRs are loaded, Randomizer has an option to enable some dungeons to be Master Quest dungeons, either by picking a set number of dungeons or a random number.
- [x] If only the Master Quest OTR is loaded, Randomizer can still be played, but with all dungeons set to Master Quest.
- [x] If only one OTR is loaded, saves that require the other or both OTR(s) present will be grayed out (i.e. MQ Saves with only vanilla OTR, or rando saves that have a mix of MQ and vanilla dungeons).
- [x] Make OTRExporter/ExtractAssets make both a normal and mq OTR if both roms are available.
  - Appimage should do this already on this PR, still need to update mac script, OTRGui and ExtractAssets. 
- [x] ~Update GetCheckFromActor, IdentifyScrubs, etc. to account for new Master Quest checks. (Proxy will be helping out with that one).~ This will be a separate PR.

If you want to test this, at the moment you will need to manually create an OTR from both a Master Quest and Vanilla rom, and rename the Master Quest one to oot-mq.otr. Before this is merged OTRExporter will be updated to do this automatically depending on rom hashes.

leggettc18 committed:
Changes OTR Extraction to have specific mq and nonmq paths.

Also updates the game to load resources according to whether or not
Master Quest or Vanilla is loaded.

leggettc18 committed:
Removes unneeded code from the last commit.

leggettc18 committed:
Fixes some weird formatting in ZRom.c

leggettc18 commented on ZAPDTR/ZAPD/ZResource.cpp:
```diff
@@ -328,7 +328,7 @@ std::string ZResource::GetSourceOutputHeader([[maybe_unused]] const std::string&
 
 		if (StringHelper::Contains(outName, "_room_") || StringHelper::Contains(outName, "_scene"))
-			prefix = "scenes";
+			prefix = "scenes/nonmq";
```
All header files are created with the nonmq path. I figure that out of all the people playing with just one OTR, there will probably be more playing with a nonmq OTR so it makes sense for this to be the default. This is swapped out for `mq` when an MQ asset needs to be loaded.

leggettc18 commented on soh/soh/OTRGlobals.cpp:
```diff
@@ -573,10 +573,13 @@ extern "C" uint32_t ResourceMgr_IsGameMasterQuest() {
+extern "C" uint32_t ResourceMgr_IsGameMasterQuest() {
+    return IsGameMasterQuest();
+}
+
```
Needed a check for Master Quest in a CPP file, this is exported only to C files, while `IsGameMasterQuest` is usable in here and available to both. Allows for keeping the existing function while keeping code DRY.

leggettc18 commented on soh/soh/OTRGlobals.cpp:
```diff
@@ -608,6 +611,17 @@ extern "C" void ResourceMgr_LoadFile(const char* resName) {
+    if (ResourceMgr_IsGameMasterQuest()) {
+        size_t pos = 0;
+        if ((pos = Path.find("/nonmq/", 0)) != std::string::npos) {
+            Path.replace(pos, 7, "/mq/");
```
This function intercepts most resource loads. Doesn't get called if MQ is not needed, and also no replace is attempted if nonmq isn't in the string at all.

leggettc18 commented on soh/soh/OTRGlobals.cpp:
```diff
@@ -673,51 +687,55 @@ extern "C" uint16_t ResourceMgr_LoadTexHeightByName(char* texPath);
+        if (ResourceMgr_IsGameMasterQuest()) {
+            size_t pos = 0;
+            if ((pos = Path.find("/nonmq/", 0)) != std::string::npos) {
+                Path.replace(pos, 7, "/mq/");
```
Needed to also intercept for MQ here, as the `LoadTexByName` by function is in libultraship, so shouldn't modify it for MQ specifically.

leggettc18 commented on soh/soh/OTRGlobals.cpp:
```diff
@@ -549,7 +549,7 @@ extern "C" uint32_t ResourceMgr_GetGameVersion()
 }
 
-extern "C" uint32_t ResourceMgr_IsGameMasterQuest() {
+uint32_t IsGameMasterQuest() {
```
This is a new function technically, but due to where I put it and it just being an extraction of the logic previously in `ResourceMgr_IsGameMasterQuest`, it doesn't appear as a diff other than the signature changing.

leggettc18 committed:
Loads oot-mq.otr and patches oot.otr on top, if both are present.

If only one or the other are present, it becomes the only and main OTR.

leggettc18 committed:
Adds ImGui Logic for whether or an MQ Checkbox.

Checkbox checked only specifies whether new saves should be MQ or not.
Checkbox is disabled or force-enabled according to which OTRs are loaded.
Also as a necessity includes tracking what game versions have been loaded
from the OTRs.

leggettc18 committed:
Adds MQ settings logic for Randomizer's ImGui menu.

leggettc18 committed:
Writes Master Quest dungeons to the spoiler log

leggettc18 committed:
Loads MQ Dungeons from spoiler, persists in save, and loads when appropriate.

leggettc18 committed:
Adds logic to prevent loading or creating incompatible rando saves.

leggettc18 commented:
Ok, seems like most of the logic surrounding dual OTR, creating rando saves with mq dungeons, and ensuring we can't load invalid saves is done. Now we just need to ensure that MQ's rando checks are all getting accounted for.

briaguya0 commented:
wow! that was fast!

leggettc18 commented:
Oh I'm sure there will be plenty of review changes and bugs that need to be figured out. Plus I'm not gonna have much time to work on it this weekend so it'll still probably be a bit before this is ready to take off of draft.

leggettc18 committed:
Fixdes some linux build issues and new rando save issues

leggettc18 committed:
Makes appimage create both vanilla and mq otrs

If either rom is present, it makes the corresponding OTR. If both are present,
it will make both. If one OTR is present but both roms are present, it will
create the missing OTR.

puddydoo commented:
I'm thinking about how ship will one day support multiple different versions

puddydoo commented:
I'm thinking about how there will one day be support for more different roms that have other things that aren't in the currently supported roms like Japanese and Chinese language, 1.0 fire temple music/gerudo symbol, etc. I wonder if it might make more sense to generate multiple OTRs from each ROM, one with all the data that is shared between all versions and one with version-specific data.

leggettc18 commented:
Eventually I believe the plan is that there will be more general support for loading a sequence of OTRs as your "base" once more roms are extractable.

leggettc18 committed:
Makes it so a randomized save file will not be marked as MQ.

Kenix3 commented on libultraship/libultraship/Archive.cpp:
```diff
@@ -320,8 +323,64 @@ namespace Ship {
+        auto t = LoadFile("version", false);
+        if (!t->bHasLoadError)
+        {
+            uint32_t gameVersion = (*((uint32_t *)t->buffer.get()));
```
This should probably use a binary reader to get the data so that we can have endian independence with this information.

leggettc18 replied:
OK, I wasn't familiar with how we normally got the version so I copied that from somewhere else. I should be able to use the BinaryReader in the ResourceManager yeah? Or at least a similar implementation? I'll look into it.

leggettc18 replied:
Actually I guess the BinaryReader is part of ZAPD, but I should be able to import the header file like ResourceMgr does.

leggettc18 replied:
Hmm, trying to use the Binary Reader already in ZAPD, but now it won't build on linux due to undefined references, but I'm not sure why since other files in libultraship are already using it just fine.

leggettc18 replied:
Ok, got that BinaryReader that will work implemented correctly, but I also discovered that the version file doesn't have a header with it's Endianness in it, so I will need to add that before this is truly complete.

leggettc18 replied:
Ok, version file has a byte representing it's Endianness before the actual version now.

Kenix3 commented on libultraship/libultraship/Archive.h:
```diff
@@ -37,11 +37,13 @@ namespace Ship
 		bool Unload();
 	private:
 		std::string MainPath;
+        std::string BasePath;
```
I'm not sure that we need a basePath. The other archives can just be loaded as a patch.

leggettc18 committed:
Merge branch 'heads/develop' into rando-mq

leggettc18 committed:
Merge remote-tracking branch 'upstream/develop' into rando-mq

leggettc18 committed:
Refactors to load all OTRs from MainPath or a specific list.

Also adds the ability to take a std::unordered_set of hashes to
validate each OTR's version file against.

leggettc18 committed:
Fixes a syntax error

leggettc18 commented:
I will be stripping out the ImGui randomizer bits and re-disabling MQ Rando so the libultraship changes and switching between MQ and Vanilla via an ImGui checkbox can be merged sooner. Updating rando checks to work with MQ dungeons will happen in a followup PR after this. Still need to update OTRGui, the mac script, and the ExtractAssets cmake target and Python script to do two OTRs if applicable.

leggettc18 committed:
Makes ExtractAssets output Vanilla and MQ OTRs if both roms are present

leggettc18 committed:
Fixes asset generation bug.

leggettc18 committed:
Partially working fix for dual OTR extract_assets

Currently the cmake ExtractAssets target will return with a 1 if you
only end up exporting one type of OTR isntead of both. Haven't found
a great way to only attempt to copy a file if it exists from within
cmake. It does actually correctly copy the OTR that is generated,
despite the error from copying the other one.

Pushing as is for now but will keep investigating.

leggettc18 committed:
Adds oot-mq.otr to the gitignore.

Kenix3 commented on libultraship/libultraship/ResourceMgr.cpp:
```diff
@@ -19,7 +20,19 @@ namespace Ship {
+    {
+        OTR = std::make_shared<Archive>(OTRFiles, PatchesPath, ValidHashes, false);
+
+        gameVersion = OOT_UNKNOWN;
```
Should probably add a new unknown enum value that doesn't imply OOT at all.

leggettc18 replied:
I don't think I actually changed that line (though my editor may have subtly changed some formatting at some point), and the macros for the various OOT hashes were already in LUS and probably shouldn't be, but I'll rename this one to just UNKNOWN for now.

leggettc18 replied:
Done.

Kenix3 commented on libultraship/libultraship/ResourceMgr.h:
```diff
@@ -17,8 +17,9 @@ namespace Ship {
-		ResourceMgr(std::shared_ptr<Window> Context, const std::string& MainPath, const std::string& PatchesPath);
-		~ResourceMgr();
+		ResourceMgr(std::shared_ptr<Window> Context, const std::string& MainPath, const std::string& PatchesPath, const std::unordered_set<uint32_t> &ValidHashes);
+        ResourceMgr(std::shared_ptr<Window> Context, const std::vector<std::string> &OTRFiles, const std::string &PatchesPath, const std::unordered_set<uint32_t> &ValidHashes);
```
This doesn't need to take in patches path anymore.

leggettc18 replied:
Handled in a recent commit.

Kenix3 commented on libultraship/libultraship/Archive.h:
```diff
@@ -21,14 +22,15 @@ namespace Ship
-		Archive(const std::string& MainPath, const std::string& PatchesPath, bool enableWriting, bool genCRCMap = true);
-		~Archive();
+        Archive(const std::string &MainPath, const std::string &PatchesPath, const std::unordered_set<uint32_t> &ValidHashes, bool enableWriting, bool genCRCMap = true);
+        Archive(const std::vector<std::string> &OTRFiles, const std::string &PatchesPath, const std::unordered_set<uint32_t> &ValidHashes, bool enableWriting, bool genCRCMap = true);
```
This probably no longer needs patches path.

leggettc18 replied:
Would we not still want mods to be imported from the `/mods` folder? If so I think we would still need that, unless we want to hardcode the presence of a `/mods` folder?

Kenix3 replied:
It just seems weird that we have a vector of OTRs and then a mods folder. I think the /mods folder should be something done in SoH at this point rather than LUS.

leggettc18 replied:
Oh you only want Patches Path removed from the one that gets the vector of OTRs? That makes much more sense, I'll fix that shortly.

Kenix3 replied:
Yeah it makes sense to only send LUS a list of archives and let SOH itself deal with the patches path and what archives and in what order to send LUS.

leggettc18 replied:
Should be handled now.

leggettc18 committed:
Makes ExtractAssets not fail on only one rom/OTR.

leggettc18 committed:
Removes PatchesPath from the constructors requiring OTRFiles vector.

leggettc18 committed:
Renames OOT_UNKNOWN to just UNKNOWN to remove OOT specific reference.

leggettc18 committed:
Merge branch 'develop' of github.com:HarbourMasters/Shipwright into rando-mq

leggettc18 committed:
Removes randomizing MQ Dungeons and re-disables MQ rando.

Doing this so the PR can get merged quicker with just the Dual OTR
support and won't need to wait on rando logic to be updated. That
will happen in another PR directly after the merge.

garrettjoecox committed:
Update mac startup script for dual otr

garrettjoecox committed:
Update soh/macosx/soh-macos.sh

garrettjoecox committed:
Update soh/macosx/soh-macos.sh

garrettjoecox committed:
Update soh/macosx/soh-macos.sh

leggettc18 committed:
Merge pull request #5 from garrettjoecox/update-mac-script

Update mac startup script for dual otr

leggettc18 committed:
Implements new BinaryReader to fix Linux build issue.

BinaryReader itself comes from https://github.com/Moneyl/BinaryTools
I added a wrapper to adapt it to the ABI from ZAPD's Binary Reader and
add Endianness checking. I also had to copy a handful of other bits and
pieces from ZAPD to make it all function as expected.

leggettc18 committed:
A few edits to the updatream BinaryReader to compile it on Linux.

leggettc18 committed:
Adds the Endianness to the first byte of the version file.

leggettc18 committed:
Merge branch 'rando-mq' of github.com:leggettc18/Shipwright; branch 'develop' of github.com:HarbourMasters/Shipwright into rando-mq

leggettc18 committed:
Fixes Jenkins

leggettc18 committed:
Merge branch 'develop' of github.com:HarbourMasters/Shipwright into rando-mq

Kenix3 commented on libultraship/libultraship/Archive.cpp:
```diff
@@ -55,15 +66,19 @@ namespace Ship {
 	}
 
-	std::shared_ptr<File> Archive::LoadFile(const std::string& filePath, bool includeParent, std::shared_ptr<File> FileToLoad) {
+	std::shared_ptr<File> Archive::LoadFile(const std::string& filePath, bool includeParent, std::shared_ptr<File> FileToLoad, HANDLE mpqHandle) {
```
Why pass in mpq handle here?

leggettc18 replied:
Previously LoadFile would only ever load from mainMpq, I added this so that PushGameVersion can specify which MPQ file it gets the version file from a different MPQ file if needed. The workflow is, after loading the Main MPQ file, it loads all the other OTRs as patches via `LoadPatchMPQ`. `LoadPatchMPQ` opens the archive first with it's own `mpqHandle` and then checks it's version file and pushes that to the list of rom hashes that were loaded. So it passes the `patchMpqHandle` to `PushGameVersion` which then pushes it to `LoadFile`. I believe I have it set up so that if you don't provide an `mpqHandle` it will just use the main mpq handle.

Kenix3 replied:
Do we need to read from a specific otr when all are loaded as patches? To get the master quest version of a scene you just change the string reference. The patching system should take care of what data is loaded, not passing in a handle. If you need to change out OTRs the the archive needs to be reloaded.

leggettc18 replied:
Well this functionality is only ever used to read the version file _before_ patching so it can store it in a set that any game using libultraship can read later. It's set up with defaults such that you can call LoadFile without providing a handle and it will load from the mainMPQ handle just like before. If you really want it gone I can just call the stormlib functions directly in `PushGameVersion` instead, I was just avoiding code duplication and needed a way to load from a different OTR for that particular case. I didn't think the option to load from a specific mpqHandle would be unwelcome.

Kenix3 replied:
If the intention is to load the version file then we need a private overloaded method that takes in the handle. The public one calls the private one with the handle as null.

leggettc18 replied:
Works for me, done in latest commit.

Kenix3 commented on libultraship/libultraship/BinaryReader.cpp:
```diff
@@ -0,0 +1,211 @@
+    return result;
+}
+
+float Ship::BinaryReader::ReadSingle()
```
```suggestion
float Ship::BinaryReader::ReadFloat()
```

leggettc18 replied:
I can change that but I did it that way because I was having my wrapper mimic the previous API which had ReadSingle instead of Float for some reason.

leggettc18 replied:
For this and the Byte->Int8 case, do you want me to go through and change all the calls to ReadSingle that were already there? Or just add the new function definition so that it can be used in the future?

Kenix3 replied:
Normally singles are called floats and int8 and uint8 is more consistent, so I say we update everything to match that.

leggettc18 replied:
Alright, most recent commit should do this for Floats and Int8s. I hadn't changed UByte to UInt8 yet but I'll do that as well in a minute.

Kenix3 commented on libultraship/libultraship/BinaryReader.cpp:
```diff
@@ -0,0 +1,211 @@
+    return Color3b();
+}
+
+std::string Ship::BinaryReader::ReadString()
```
Should probably also add a method to read null terminated strings.

leggettc18 replied:
Easy enough, the underlying reader already has a method for exactly this. Should I call it `ReadCString` or the lengthier `ReadNullTerminatedString`?

Kenix3 replied:
I like NullString or CString

leggettc18 replied:
OK, most recent commit has it called `ReadNullTerminatedString`. If you're OK with CString though I'll change it to that.

Kenix3 replied:
Either works.

leggettc18 replied:
Done in most recent commit.

Kenix3 commented on libultraship/libultraship/BinaryReader.h:
```diff
@@ -0,0 +1,59 @@
+    void Read(int32_t length);
+    void Read(char *buffer, int32_t length);
+    char ReadChar();
+    int8_t ReadByte();
```
```suggestion
    int8_t ReadInt8();
```
For consistency

leggettc18 replied:
Same as the Single one, done to mimic the previous ABI.

leggettc18 replied:
Should be fixed in recent commit.

Kenix3 commented on libultraship/libultraship/BitConverter.h:
```diff
@@ -0,0 +1,186 @@
+#pragma once
```
Does this class need to be in LUS? Seems to me like the new full featured binary reader/writer does everything this was needed for.

leggettc18 replied:
I'll take a closer look, I think it has the Endianness enum in it but I can probably just strip that out and put it somewhere else.

leggettc18 replied:
On closer look, this is needed to handle switching Endianness, all of the functions in there are used in the new BinaryReader wrapper and it makes sense to be a separate file because the BinaryWriter (which we'll probably want but is unused currently in libultraship) will also need it as well.

Kenix3 replied:
If they're used in the new wrapper then it's fine. I thought you were using intrinsics in the new wrapper.

leggettc18 replied:
Yeah I used this in the new wrapper so the places in existing code that set endianness values would still work the same way. It does appear that this class in and of itself is using intrinsics though.

leggettc18 committed:
Addresses some of Kenix's comments

leggettc18 committed:
Renames `ReadNullTerminatedString` to `ReadCString`

leggettc18 committed:
Refactors Archive::LoadFile into a private method with more arguments.

Kenix3 reviewed (commented):
Two small things left. First is that in LUS braces go on the same line as the function definition.

Second I'm not sure BitConverter needs to be in LUS. Will ping you on Discord.

leggettc18 committed:
Removes BitConverter and extends existing endianness.h instead.

leggettc18 committed:
Fixes an endianness issue with the version file.

briaguya0 commented on ZAPDTR/ZAPD/ZResource.cpp:
```diff
@@ -342,8 +342,9 @@ std::string ZResource::GetSourceOutputHeader([[maybe_unused]] const std::string&
-		if (prefix != "")
+		if (prefix != "") {
 			str += StringHelper::Sprintf("#define d%s \"__OTR__%s/%s/%s\"", name.c_str(), prefix.c_str(), outName.c_str(), nameStr.c_str());
+        }
```
tiny nitpick, tabs vs spaces

briaguya0 commented on ZAPDTR/ZAPD/ZRom.h:
```diff
@@ -11,6 +11,7 @@ class ZRom
 	ZRom(std::string romPath);
 
 	std::vector<uint8_t> GetFile(std::string fileName);
+    bool IsMQ();
```
tab v spaces

briaguya0 commented on soh/soh/Enhancements/randomizer/3drando/settings.cpp:
```diff
@@ -2535,6 +2535,13 @@ namespace Settings {
     } else {
         GanonsTrialsCount.SetSelectedIndex(cvarSettings[RSK_TRIAL_COUNT]);
     }
+    if (cvarSettings[RSK_RANDOM_MQ_DUNGEONS] == 2) {
+        MQDungeonCount.SetSelectedIndex(13);
+    } else if (cvarSettings[RSK_RANDOM_MQ_DUNGEONS] == 0) {
+        MQDungeonCount.SetSelectedIndex(0);
+    } else {
+        MQDungeonCount.SetSelectedIndex(cvarSettings[RSK_MQ_DUNGEON_COUNT]);
+    }
```
there's a [PR out](https://github.com/HarbourMasters/Shipwright/pull/1761) for moving away from magic numbers for RSKs to using enum values, it hasn't been merged yet but i figured it's worth adding a note to so we don't forget to add these in there

briaguya0 commented on soh/soh/OTRGlobals.h:
```diff
@@ -25,13 +25,20 @@ class OTRGlobals
 
 #ifndef __cplusplus
-void InitOTR(void);
+    void InitOTR(void);
```
tiny nitpick: spacing

briaguya0 reviewed (approved):
couple tiny comments about formatting in a couple places but overall it's looking great!

:shipit:

briaguya0 commented:
tested and working on linux! once this has been tested on mac let's get this merged!

garrettjoecox commented:
> tested and working on linux! once this has been tested on mac let's get this merged!

Working on mac!

xoascf commented:
For some reason after this PR, OTR mods no longer work for me 😕

briaguya0 commented:
> For some reason after this PR, OTR mods no longer work for me confused

could you be more specific? asset paths have changed so mods will need to be updated to support this

xoascf commented:
> > For some reason after this PR, OTR mods no longer work for me confused
> 
> could you be more specific? asset paths have changed so mods will need to be updated to support this

Mods no longer load, or at least do not make any effect or difference in-game (as with texture mods located in /mods path).

<details>
<summary>Files in my OTR</summary>
<pre>
textures/item_name_static/gDekuStickItemNameENGTex
textures/item_name_static/gDekuNutItemNameENGTex
textures/item_name_static/gBombItemNameENGTex
textures/item_name_static/gFairyBowItemNameENGTex
textures/item_name_static/gFireArrowItemNameENGTex
textures/item_name_static/gDinsFireItemNameENGTex
textures/item_name_static/gFairySlingshotItemNameENGTex
textures/item_name_static/gFairyOcarinaItemNameENGTex
textures/item_name_static/gOcarinaOfTimeItemNameENGTex
textures/item_name_static/gBombchuItemNameENGTex
textures/item_name_static/gHookshotItemNameENGTex
textures/item_name_static/gLongshotItemNameENGTex
textures/item_name_static/gIceArrowItemNameENGTex
textures/item_name_static/gFaroresWindItemNameENGTex
textures/item_name_static/gBoomerangItemNameENGTex
textures/item_name_static/gLensItemNameENGTex
textures/item_name_static/gMagicBeansItemNameENGTex
textures/item_name_static/gMegatonHammerItemNameENGTex
textures/item_name_static/gLightArrowItemNameENGTex
textures/item_name_static/gNayrusLoveItemNameENGTex
textures/item_name_static/gEmptyBottleItemNameENGTex
textures/item_name_static/gRedPotionItemNameENGTex
textures/item_name_static/gGreenPotionItemNameENGTex
textures/item_name_static/gBluePotionItemNameENGTex
textures/item_name_static/gBottledFairyItemNameENGTex
textures/item_name_static/gFishItemNameENGTex
textures/item_name_static/gFullMilkItemNameENGTex
textures/item_name_static/gRutosLetterItemNameENGTex
textures/item_name_static/gBlueFireItemNameENGTex
textures/item_name_static/gBugItemNameENGTex
textures/item_name_static/gBigPoeItemNameENGTex
textures/item_name_static/gHalfMilkItemNameENGTex
textures/item_name_static/gPoeItemNameENGTex
textures/item_name_static/gWeirdEggItemNameENGTex
textures/item_name_static/gCuccoItemNameENGTex
textures/item_name_static/gZeldasLetterItemNameENGTex
textures/item_name_static/gKeatonMaskItemNameENGTex
textures/item_name_static/gSkullMaskItemNameENGTex
textures/item_name_static/gSpookyMaskItemNameENGTex
textures/item_name_static/gBunnyHoodItemNameENGTex
textures/item_name_static/gGoronMaskItemNameENGTex
textures/item_name_static/gZoraMaskItemNameENGTex
textures/item_name_static/gGerudoMaskItemNameENGTex
textures/item_name_static/gMaskofTruthItemNameENGTex
textures/item_name_static/gSOLDOUTItemNameENGTex
textures/item_name_static/gPocketEggItemNameENGTex
textures/item_name_static/gPocketCuccoItemNameENGTex
textures/item_name_static/gCojiroItemNameENGTex
textures/item_name_static/gOddMushroomItemNameENGTex
textures/item_name_static/gOddPotionItemNameENGTex
textures/item_name_static/gPoachersSawItemNameENGTex
textures/item_name_static/gBrokenGoronsSwordItemNameENGTex
textures/item_name_static/gPrescriptionItemNameENGTex
textures/item_name_static/gEyeBallFrogItemNameENGTex
textures/item_name_static/gEyeDropsItemNameENGTex
textures/item_name_static/gClaimCheckItemNameENGTex
textures/item_name_static/gKokiriSwordItemNameENGTex
textures/item_name_static/gMasterSwordItemNameENGTex
textures/item_name_static/gGiantsKnifeItemNameENGTex
textures/item_name_static/gDekuShieldItemNameENGTex
textures/item_name_static/gHylianShieldItemNameENGTex
textures/item_name_static/gMirrorShieldItemNameENGTex
textures/item_name_static/gKokiriTunicItemNameENGTex
textures/item_name_static/gGoronTunicItemNameENGTex
textures/item_name_static/gZoraTunicItemNameENGTex
textures/item_name_static/gKokiriBootsItemNameENGTex
textures/item_name_static/gIronBootsItemNameENGTex
textures/item_name_static/gHoverBootsItemNameENGTex
textures/item_name_static/gBulletBag30ItemNameENGTex
textures/item_name_static/gBulletBag40ItemNameENGTex
textures/item_name_static/gBulletBag50ItemNameENGTex
textures/item_name_static/gQuiver30ItemNameENGTex
textures/item_name_static/gQuiver40ItemNameENGTex
textures/item_name_static/gQuiver50ItemNameENGTex
textures/item_name_static/gBombBag20ItemNameENGTex
textures/item_name_static/gBombBag30ItemNameENGTex
textures/item_name_static/gBombBag40ItemNameENGTex
textures/item_name_static/gGoronsBraceletItemNameENGTex
textures/item_name_static/gSilverGauntletsItemNameENGTex
textures/item_name_static/gGoldenGauntletsItemNameENGTex
textures/item_name_static/gSilverScaleItemNameENGTex
textures/item_name_static/gGoldenScaleItemNameENGTex
textures/item_name_static/gBrokenGiantsKnifeItemNameENGTex
textures/item_name_static/gMinuetOfForestItemNameENGTex
textures/item_name_static/gBoleroOfFireItemNameENGTex
textures/item_name_static/gSerenadeOfWaterItemNameENGTex
textures/item_name_static/gRequiemOfSpiritItemNameENGTex
textures/item_name_static/gNocturneOfShadowItemNameENGTex
textures/item_name_static/gPreludeOfLightItemNameENGTex
textures/item_name_static/gZeldasLullabyItemNameENGTex
textures/item_name_static/gEponasSongItemNameENGTex
textures/item_name_static/gSariasSongItemNameENGTex
textures/item_name_static/gSunsSongItemNameENGTex
textures/item_name_static/gSongOfTimeItemNameENGTex
textures/item_name_static/gSongOfStormsItemNameENGTex
textures/item_name_static/gForestMedallionItemNameENGTex
textures/item_name_static/gFireMedallionItemNameENGTex
textures/item_name_static/gWaterMedallionItemNameENGTex
textures/item_name_static/gSpiritMedallionItemNameENGTex
textures/item_name_static/gShadowMedallionItemNameENGTex
textures/item_name_static/gLightMedallionItemNameENGTex
textures/item_name_static/gKokiriEmeraldItemNameENGTex
textures/item_name_static/gGoronsRubyItemNameENGTex
textures/item_name_static/gZorasSapphireItemNameENGTex
textures/item_name_static/gStoneofAgonyItemNameENGTex
textures/item_name_static/gGerudosCardItemNameENGTex
textures/item_name_static/gGoldSkulltulaItemNameENGTex
textures/item_name_static/gPieceOfHeartItemNameENGTex
textures/item_name_static/gUnusedBigKeyItemNameENGTex
textures/item_name_static/gCompassItemNameENGTex
textures/item_name_static/gDungeonMapItemNameENGTex
textures/item_name_static/gBiggoronsSwordItemNameENGTex
textures/do_action_static/gAttackDoActionENGTex
textures/do_action_static/gCheckDoActionENGTex
textures/do_action_static/gEnterDoActionENGTex
textures/do_action_static/gReturnDoActionENGTex
textures/do_action_static/gOpenDoActionENGTex
textures/do_action_static/gJumpDoActionENGTex
textures/do_action_static/gDecideDoActionENGTex
textures/do_action_static/gDiveDoActionENGTex
textures/do_action_static/gFasterDoActionENGTex
textures/do_action_static/gThrowDoActionENGTex
textures/do_action_static/gClimbDoActionENGTex
textures/do_action_static/gDropDoActionENGTex
textures/do_action_static/gDownDoActionENGTex
textures/do_action_static/gSaveDoActionENGTex
textures/do_action_static/gSpeakDoActionENGTex
textures/do_action_static/gNextDoActionENGTex
textures/do_action_static/gGrabDoActionENGTex
textures/do_action_static/gStopDoActionENGTex
textures/do_action_static/gPutAwayDoActionENGTex
textures/do_action_static/gReelDoActionENGTex
textures/icon_item_dungeon_static/gDungeonMap1FButtonTex
textures/icon_item_dungeon_static/gDungeonMap2FButtonTex
textures/icon_item_dungeon_static/gDungeonMap3FButtonTex
textures/icon_item_dungeon_static/gDungeonMap4FButtonTex
textures/icon_item_dungeon_static/gDungeonMap5FButtonTex
textures/icon_item_dungeon_static/gDungeonMap6FButtonTex
textures/icon_item_dungeon_static/gDungeonMap7FButtonTex
textures/icon_item_dungeon_static/gDungeonMap8FButtonTex
textures/icon_item_dungeon_static/gDungeonMapB1ButtonTex
textures/icon_item_dungeon_static/gDungeonMapB2ButtonTex
textures/icon_item_dungeon_static/gDungeonMapB3ButtonTex
textures/icon_item_dungeon_static/gDungeonMapB4ButtonTex
textures/icon_item_dungeon_static/gDungeonMapB5ButtonTex
textures/icon_item_dungeon_static/gDungeonMapB6ButtonTex
textures/icon_item_dungeon_static/gDungeonMapB7ButtonTex
textures/icon_item_dungeon_static/gDungeonMapB8ButtonTex
textures/icon_item_gameover_static/gGameOverP1Tex
textures/icon_item_gameover_static/gGameOverP2Tex
textures/icon_item_gameover_static/gGameOverP3Tex
textures/icon_item_gameover_static/gContinuePlayingENGTex
textures/map_name_static/gHauntedWastelandPointNameENGTex
textures/map_name_static/gGerudosFortressPointNameENGTex
textures/map_name_static/gGerudoValleyPointNameENGTex
textures/map_name_static/gHyliaLakesidePointNameENGTex
textures/map_name_static/gLonLonRanchPointNameENGTex
textures/map_name_static/gMarketPointNameENGTex
textures/map_name_static/gHyruleFieldPointNameENGTex
textures/map_name_static/gDeathMountainPointNameENGTex
textures/map_name_static/gKakarikoVillagePointNameENGTex
textures/map_name_static/gLostWoodsPointNameENGTex
textures/map_name_static/gKokiriForestPointNameENGTex
textures/map_name_static/gZorasDomainPointNameENGTex
textures/map_name_static/gHyruleFieldPositionNameENGTex
textures/map_name_static/gKakarikoVillagePositionNameENGTex
textures/map_name_static/gGraveyardPositionNameENGTex
textures/map_name_static/gZorasRiverPositionNameENGTex
textures/map_name_static/gKokiriForestPositionNameENGTex
textures/map_name_static/gSacredForestMeadowPositionNameENGTex
textures/map_name_static/gLakeHyliaPositionNameENGTex
textures/map_name_static/gZorasDomainPositionNameENGTex
textures/map_name_static/gZorasFountainPositionNameENGTex
textures/map_name_static/gGerudoValleyPositionNameENGTex
textures/map_name_static/gLostWoodsPositionNameENGTex
textures/map_name_static/gDesertColossusPositionNameENGTex
textures/map_name_static/gGerudosFortressPositionNameENGTex
textures/map_name_static/gHauntedWastelandPositionNameENGTex
textures/map_name_static/gMarketPositionNameENGTex
textures/map_name_static/gHyruleCastlePositionNameENGTex
textures/map_name_static/gDeathMountainTrailPositionNameENGTex
textures/map_name_static/gDeathMountainCraterPositionNameENGTex
textures/map_name_static/gGoronCityPositionNameENGTex
textures/map_name_static/gLonLonRanchPositionNameENGTex
textures/map_name_static/gQuestionMarkPositionNameENGTex
textures/map_name_static/gGanonsCastlePositionNameENGTex
textures/g_pn_01/gForestTempleTitleCardENGTex
textures/g_pn_02/gShadowTempleTitleCardENGTex
textures/g_pn_03/gFireTempleTitleCardENGTex
textures/g_pn_04/gWaterTempleTitleCardENGTex
textures/g_pn_05/gSpiritTempleTitleCardENGTex
textures/g_pn_06/gDekuTreeTitleCardENGTex
textures/g_pn_07/gJabuJabuTitleCardENGTex
textures/g_pn_08/gDodongosCavernTitleCardENGTex
textures/g_pn_09/gInsideGanonsCastleTitleCardENGTex
textures/g_pn_10/gIceCavernTitleCardENGTex
textures/g_pn_11/gGERudoTrainingGroundTitleCardENGTex
textures/g_pn_12/gCastleCourtyardTitleCardENGTex
textures/g_pn_13/gGreatFairysFountainTitleCardENGTex
textures/g_pn_14/gChamberOfTheSagesTitleCardENGTex
textures/g_pn_15/gShootingGalleryTitleCardENGTex
textures/g_pn_16/gTempleOfTimeTitleCardENGTex
textures/g_pn_17/gMarketTitleCardENGTex
textures/g_pn_18/gBackAlleyTitleCardENGTex
textures/g_pn_19/gKokiriShopTitleCardENGTex
textures/g_pn_20/gGoronShopTitleCardENGTex
textures/g_pn_21/gZoraShopTitleCardENGTex
textures/g_pn_22/gHouseOfSkulltulaTitleCardENGTex
textures/g_pn_23/gBazaarTitleCardENGTex
textures/g_pn_24/gPotionShopTitleCardENGTex
textures/g_pn_25/gGravekeepersHutTitleCardENGTex
textures/g_pn_26/gLakesideLaboratoryTitleCardENGTex
textures/g_pn_27/gHyruleFieldTitleCardENGTex
textures/g_pn_28/gKakarikoVillageTitleCardENGTex
textures/g_pn_29/gGraveyardTitleCardENGTex
textures/g_pn_30/gZorasRiverTitleCardENGTex
textures/g_pn_31/gKokiriForestTitleCardENGTex
textures/g_pn_32/gLakeHyliaTitleCardENGTex
textures/g_pn_33/gZorasDomainTitleCardENGTex
textures/g_pn_34/gZorasFountainTitleCardENGTex
textures/g_pn_35/gGERudoValleyTitleCardENGTex
textures/g_pn_36/gLostWoodsTitleCardENGTex
textures/g_pn_37/gHauntedWastelandTitleCardENGTex
textures/g_pn_38/gHyruleCastleTitleCardENGTex
textures/g_pn_39/gDeathMountainTrailTitleCardENGTex
textures/g_pn_40/gDeathMountainCraterTitleCardENGTex
textures/g_pn_41/gGoronCityTitleCardENGTex
textures/g_pn_42/gLonLonRanchTitleCardENGTex
textures/g_pn_43/gGanonsCastleTitleCardENGTex
textures/g_pn_44/gRoyalFamilysTombTitleCardENGTex
textures/g_pn_45/gFairysFountainTitleCardENGTex
textures/g_pn_46/gFishingPondTitleCardENGTex
textures/g_pn_47/gBombchuBowlingAlleyCardENGTex
textures/g_pn_48/gStableTitleCardENGTex
textures/g_pn_49/gThievesHideoutTitleCardENGTex
textures/g_pn_50/gHappyMaskShopTitleCardENGTex
textures/g_pn_51/gTreasureBoxShopTitleCardENGTex
textures/g_pn_52/gSacredForestMeadowTitleCardENGTex
textures/g_pn_53/gGERudosFortressTitleCardENGTex
textures/g_pn_54/gBottomOfTheWellTitleCardENGTex
textures/g_pn_55/gDesertColossusTitleCardENGTex
textures/g_pn_56/gBombchuShopTitleCardENGTex
textures/g_pn_57/gQuestionMarkTitleCardENGTex
textures/nes_font_static/gMsgChar20SpaceTex
textures/nes_font_static/gMsgChar21ExclamationMarkTex
textures/nes_font_static/gMsgChar22QuotationMarkTex
textures/nes_font_static/gMsgChar23NumberSignTex
textures/nes_font_static/gMsgChar24DollarSignTex
textures/nes_font_static/gMsgChar25PercentSignTex
textures/nes_font_static/gMsgChar26AmpersandTex
textures/nes_font_static/gMsgChar27ApostropheTex
textures/nes_font_static/gMsgChar28LeftParenthesesTex
textures/nes_font_static/gMsgChar29RightParenthesesTex
textures/nes_font_static/gMsgChar2AAsteriskTex
textures/nes_font_static/gMsgChar2BPlusSignTex
textures/nes_font_static/gMsgChar2CCommaTex
textures/nes_font_static/gMsgChar2DHyphenMinusTex
textures/nes_font_static/gMsgChar2EFullStopTex
textures/nes_font_static/gMsgChar2FSolidusTex
textures/nes_font_static/gMsgChar30Digit0Tex
textures/nes_font_static/gMsgChar31Digit1Tex
textures/nes_font_static/gMsgChar32Digit2Tex
textures/nes_font_static/gMsgChar33Digit3Tex
textures/nes_font_static/gMsgChar34Digit4Tex
textures/nes_font_static/gMsgChar35Digit5Tex
textures/nes_font_static/gMsgChar36Digit6Tex
textures/nes_font_static/gMsgChar37Digit7Tex
textures/nes_font_static/gMsgChar38Digit8Tex
textures/nes_font_static/gMsgChar39Digit9Tex
textures/nes_font_static/gMsgChar3AColonTex
textures/nes_font_static/gMsgChar3BSemicolonTex
textures/nes_font_static/gMsgChar3CLessThanSignTex
textures/nes_font_static/gMsgChar3DEqualsSignTex
textures/nes_font_static/gMsgChar3EGreaterThanSignTex
textures/nes_font_static/gMsgChar3FQuestionMarkTex
textures/nes_font_static/gMsgChar40CommercialAtTex
textures/nes_font_static/gMsgChar41LatinCapitalLetterATex
textures/nes_font_static/gMsgChar42LatinCapitalLetterBTex
textures/nes_font_static/gMsgChar43LatinCapitalLetterCTex
textures/nes_font_static/gMsgChar44LatinCapitalLetterDTex
textures/nes_font_static/gMsgChar45LatinCapitalLetterETex
textures/nes_font_static/gMsgChar46LatinCapitalLetterFTex
textures/nes_font_static/gMsgChar47LatinCapitalLetterGTex
textures/nes_font_static/gMsgChar48LatinCapitalLetterHTex
textures/nes_font_static/gMsgChar49LatinCapitalLetterITex
textures/nes_font_static/gMsgChar4ALatinCapitalLetterJTex
textures/nes_font_static/gMsgChar4BLatinCapitalLetterKTex
textures/nes_font_static/gMsgChar4CLatinCapitalLetterLTex
textures/nes_font_static/gMsgChar4DLatinCapitalLetterMTex
textures/nes_font_static/gMsgChar4ELatinCapitalLetterNTex
textures/nes_font_static/gMsgChar4FLatinCapitalLetterOTex
textures/nes_font_static/gMsgChar50LatinCapitalLetterPTex
textures/nes_font_static/gMsgChar51LatinCapitalLetterQTex
textures/nes_font_static/gMsgChar52LatinCapitalLetterRTex
textures/nes_font_static/gMsgChar53LatinCapitalLetterSTex
textures/nes_font_static/gMsgChar54LatinCapitalLetterTTex
textures/nes_font_static/gMsgChar55LatinCapitalLetterUTex
textures/nes_font_static/gMsgChar56LatinCapitalLetterVTex
textures/nes_font_static/gMsgChar57LatinCapitalLetterWTex
textures/nes_font_static/gMsgChar58LatinCapitalLetterXTex
textures/nes_font_static/gMsgChar59LatinCapitalLetterYTex
textures/nes_font_static/gMsgChar5ALatinCapitalLetterZTex
textures/nes_font_static/gMsgChar5BLeftSquareBracketTex
textures/nes_font_static/gMsgChar5CYenSignTex
textures/nes_font_static/gMsgChar5DRightSquareBracketTex
textures/nes_font_static/gMsgChar5ECircumflexAccentTex
textures/nes_font_static/gMsgChar5FLowLineTex
textures/nes_font_static/gMsgChar60GraveAccentTex
textures/nes_font_static/gMsgChar61LatinSmallLetterATex
textures/nes_font_static/gMsgChar62LatinSmallLetterBTex
textures/nes_font_static/gMsgChar63LatinSmallLetterCTex
textures/nes_font_static/gMsgChar64LatinSmallLetterDTex
textures/nes_font_static/gMsgChar65LatinSmallLetterETex
textures/nes_font_static/gMsgChar66LatinSmallLetterFTex
textures/nes_font_static/gMsgChar67LatinSmallLetterGTex
textures/nes_font_static/gMsgChar68LatinSmallLetterHTex
textures/nes_font_static/gMsgChar69LatinSmallLetterITex
textures/nes_font_static/gMsgChar6ALatinSmallLetterJTex
textures/nes_font_static/gMsgChar6BLatinSmallLetterKTex
textures/nes_font_static/gMsgChar6CLatinSmallLetterLTex
textures/nes_font_static/gMsgChar6DLatinSmallLetterMTex
textures/nes_font_static/gMsgChar6ELatinSmallLetterNTex
textures/nes_font_static/gMsgChar6FLatinSmallLetterOTex
textures/nes_font_static/gMsgChar70LatinSmallLetterPTex
textures/nes_font_static/gMsgChar71LatinSmallLetterQTex
textures/nes_font_static/gMsgChar72LatinSmallLetterRTex
textures/nes_font_static/gMsgChar73LatinSmallLetterSTex
textures/nes_font_static/gMsgChar74LatinSmallLetterTTex
textures/nes_font_static/gMsgChar75LatinSmallLetterUTex
textures/nes_font_static/gMsgChar76LatinSmallLetterVTex
textures/nes_font_static/gMsgChar77LatinSmallLetterWTex
textures/nes_font_static/gMsgChar78LatinSmallLetterXTex
textures/nes_font_static/gMsgChar79LatinSmallLetterYTex
textures/nes_font_static/gMsgChar7ALatinSmallLetterZTex
textures/nes_font_static/gMsgChar7BLeftCurlyBracketTex
textures/nes_font_static/gMsgChar7CVerticalLineTex
textures/nes_font_static/gMsgChar7DRightCurlyBracketTex
textures/nes_font_static/gMsgChar7ETildeTex
textures/nes_font_static/gMsgChar7FBlankTex
textures/nes_font_static/gMsgChar80LatinCapitalLetterAWithGraveTex
textures/nes_font_static/gMsgChar81LatinCapitalLetterIWithCircumflexTex
textures/nes_font_static/gMsgChar82LatinCapitalLetterAWithCircumflexTex
textures/nes_font_static/gMsgChar83LatinCapitalLetterAWithDiaeresisTex
textures/nes_font_static/gMsgChar84LatinCapitalLetterCWithCedillaTex
textures/nes_font_static/gMsgChar85LatinCapitalLetterEWithGraveTex
textures/nes_font_static/gMsgChar86LatinCapitalLetterEWithAcuteTex
textures/nes_font_static/gMsgChar87LatinCapitalLetterEWithCircumflexTex
textures/nes_font_static/gMsgChar88LatinCapitalLetterEWithDiaeresisTex
textures/nes_font_static/gMsgChar89LatinCapitalLetterIWithDiaeresisTex
textures/nes_font_static/gMsgChar8ALatinCapitalLetterOWithCircumflexTex
textures/nes_font_static/gMsgChar8BLatinCapitalLetterOWithDiaeresisTex
textures/nes_font_static/gMsgChar8CLatinCapitalLetterUWithGraveTex
textures/nes_font_static/gMsgChar8DLatinCapitalLetterUWithCircumflexTex
textures/nes_font_static/gMsgChar8ELatinCapitalLetterUWithDiaeresisTex
textures/nes_font_static/gMsgChar8FLatinSmallLetterSharpSTex
textures/nes_font_static/gMsgChar90LatinSmallLetterAWithGraveTex
textures/nes_font_static/gMsgChar91LatinSmallLetterAWithAcuteTex
textures/nes_font_static/gMsgChar92LatinSmallLetterAWithCircumflexTex
textures/nes_font_static/gMsgChar93LatinSmallLetterAWithDiaeresisTex
textures/nes_font_static/gMsgChar94LatinSmallLetterCWithCedillaTex
textures/nes_font_static/gMsgChar95LatinSmallLetterEWithGraveTex
textures/nes_font_static/gMsgChar96LatinSmallLetterEWithAcuteTex
textures/nes_font_static/gMsgChar97LatinSmallLetterEWithCircumflexTex
textures/nes_font_static/gMsgChar98LatinSmallLetterEWithDiaeresisTex
textures/nes_font_static/gMsgChar99LatinSmallLetterIWithDiaeresisTex
textures/nes_font_static/gMsgChar9ALatinSmallLetterOWithCircumflexTex
textures/nes_font_static/gMsgChar9BLatinSmallLetterOWithDiaeresisTex
textures/nes_font_static/gMsgChar9CLatinSmallLetterUWithGraveTex
textures/nes_font_static/gMsgChar9DLatinSmallLetterUWithCircumflexTex
textures/nes_font_static/gMsgChar9ELatinSmallLetterUWithDiaeresisTex
textures/nes_font_static/gMsgChar9FButtonATex
textures/nes_font_static/gMsgCharA0ButtonBTex
textures/nes_font_static/gMsgCharA1ButtonCTex
textures/nes_font_static/gMsgCharA2ButtonLTex
textures/nes_font_static/gMsgCharA3ButtonRTex
textures/nes_font_static/gMsgCharA4ButtonZTex
textures/nes_font_static/gMsgCharA5ButtonCUpTex
textures/nes_font_static/gMsgCharA6ButtonCDownTex
textures/nes_font_static/gMsgCharA7ButtonCLeftTex
textures/nes_font_static/gMsgCharA8ButtonCRightTex
textures/nes_font_static/gMsgCharA9ZTargetSignTex
textures/nes_font_static/gMsgCharAAControlStickTex
textures/icon_item_nes_static/gPauseDekuTitleENGTex
textures/icon_item_nes_static/gPauseDodongoTitleENGTex
textures/icon_item_nes_static/gPauseJabuTitleENGTex
textures/icon_item_nes_static/gPauseForestTitleENGTex
textures/icon_item_nes_static/gPauseFireTitleENGTex
textures/icon_item_nes_static/gPauseWaterTitleENGTex
textures/icon_item_nes_static/gPauseSpiritTitleENGTex
textures/icon_item_nes_static/gPauseShadowTitleENGTex
textures/icon_item_nes_static/gPauseBotWTitleENGTex
textures/icon_item_nes_static/gPauseIceCavernTitleENGTex
textures/icon_item_nes_static/gPauseToEquipENGTex
textures/icon_item_nes_static/gPauseToDecideENGTex
textures/icon_item_nes_static/gPauseToPlayMelodyENGTex
textures/icon_item_nes_static/gPauseToSelectItemENGTex
textures/icon_item_nes_static/gPauseToMapENGTex
textures/icon_item_nes_static/gPauseToQuestStatusENGTex
textures/icon_item_nes_static/gPauseToEquipmentENGTex
textures/icon_item_nes_static/gPauseSavePromptENGTex
textures/icon_item_nes_static/gPauseSaveConfirmationENGTex
textures/icon_item_nes_static/gPauseYesENGTex
textures/icon_item_nes_static/gPauseNoENGTex
textures/icon_item_nes_static/gPauseCurrentPositionENGTex
textures/icon_item_nes_static/gPauseEquipment10ENGTex
textures/icon_item_nes_static/gPauseSelectItem00ENGTex
textures/icon_item_nes_static/gPauseSelectItem10ENGTex
textures/icon_item_nes_static/gPauseSelectItem20ENGTex
textures/icon_item_nes_static/gPauseMap10ENGTex
textures/icon_item_nes_static/gPauseQuestStatus00ENGTex
textures/icon_item_nes_static/gPauseQuestStatus10ENGTex
textures/icon_item_nes_static/gPauseQuestStatus20ENGTex
textures/icon_item_nes_static/gPauseSave10ENGTex
textures/icon_item_static/gMilkFullIconTex
textures/icon_item_static/gMilkhalfIconTex
textures/icon_item_static/gSoldOutIconTex
textures/icon_item_static/gMirrorShieldIconTex
objects/object_mag/gTitleCopyright19982003Tex
textures/title_static/gFileSelSOUNDENGTex
textures/title_static/gFileSelLTargetingENGTex
textures/title_static/gFileSelNoFileToCopyENGTex
textures/title_static/gFileSelNoFileToEraseENGTex
textures/title_static/gFileSelNoEmptyFileENGTex
textures/title_static/gFileSelFileEmptyENGTex
textures/title_static/gFileSelFileInUseENGTex
textures/title_static/gFileSelCopyWhichFileENGTex
textures/title_static/gFileSelCopyToWhichFileENGTex
textures/title_static/gFileSelAreYouSureENGTex
textures/title_static/gFileSelFileCopiedENGTex
textures/title_static/gFileSelPleaseSelectAFileENGTex
textures/title_static/gFileSelOpenThisFileENGTex
textures/title_static/gFileSelEraseWhichFileENGTex
textures/title_static/gFileSelAreYouSure2ENGTex
textures/title_static/gFileSelFileErasedENGTex
textures/title_static/gFileSelOptionsENGTex
textures/title_static/gFileSelNameENGTex
textures/title_static/gFileSelSurroundENGTex
textures/title_static/gFileSelHeadsetENGTex
textures/title_static/gFileSelMonoENGTex
textures/title_static/gFileSelStereoENGTex
textures/title_static/gFileSelSwitchENGTex
textures/title_static/gFileSelHoldENGTex
textures/title_static/gFileSelControlsENGTex
textures/title_static/gFileSelCheckBrightnessENGTex
textures/title_static/gFileSelCopyButtonENGTex
textures/title_static/gFileSelFile1ButtonENGTex
textures/title_static/gFileSelFile2ButtonENGTex
textures/title_static/gFileSelFile3ButtonENGTex
textures/title_static/gFileSelYesButtonENGTex
textures/title_static/gFileSelEraseButtonENGTex
textures/title_static/gFileSelOptionsButtonENGTex
textures/title_static/gFileSelQuitButtonENGTex
textures/title_static/gFileSelENDButtonENGTex
textures/title_static/gFileSelDISKButtonTex
objects/object_bv/gBarinadeTitleCardTex
objects/object_sst/gBongoTitleCardTex
objects/object_ganon/gGanondorfTitleCardTex
objects/object_ganon2/gGanonTitleCardTex
objects/object_goma/gGohmaTitleCardTex
objects/object_kingdodongo/gKingDodongoTitleCardTex
objects/object_mo/gMorphaTitleCardTex
objects/object_fhg/gPhantomGanonTitleCardTex
objects/object_tw/gTwinrovaTitleCardTex
objects/object_fd/gVolvagiaBossTitleCardTex
overlays/ovl_End_Title/sTheLegendOfZeldaTex
overlays/ovl_End_Title/sOcarinaOfTimeTex
overlays/ovl_End_Title/sTheEndTex
overlays/ovl_End_Title/sPresentedByTex
objects/object_mir_ray/gShieldBeamImageGerudoSymbolNewTex
objects/object_gi_gerudo/object_gi_gerudoTex_000000
objects/object_link_boy/gLinkAdultMirrorShieldUpperDesignTex
objects/object_gi_shield_3/object_gi_shield_3Tex_000000
audio/sequences/042_Fire_Temple
nes_message_data_static
table

</pre>
</details>

leggettc18 commented:
Hello @xoascf, can you try this build and see if your mod will load? https://github.com/HarbourMasters/Shipwright/pull/1785

xoascf commented:
> Hello @xoascf, can you try this build and see if your mod will load? #1785

Yeah, it now works better than ever before, thank you!

![imagen](https://user-images.githubusercontent.com/28744392/196247928-4805270b-0b6a-4842-aa50-b35bb9c301eb.png)
![imagen](https://user-images.githubusercontent.com/28744392/196248019-bd21e775-e339-455a-8ff2-5c76dd728368.png)
![imagen](https://user-images.githubusercontent.com/28744392/196247955-76a3847c-0afb-4978-9e2e-1dd0ba7cb95c.png)
